packages/expo/src/winter/fetch/FetchResponse.ts

```diff
@@ -0,0 +1,111 @@
+import { ReadableStream } from 'web-streams-polyfill';
+
+import { NativeResponse } from './NativeRequest';
+
+/**
+ * A response implementation for the `fetch.Response` API.
+ */
+export class FetchResponse implements Response {
```

Comment:
It's just a wrapper for the `NativeResponse` to add some more functionality, right?
Most getters (e.g. `bodyUsed`, `status`) are just returning the same property from the native response.

If I understand correctly, we want to add some functionality to a natively-defined JS class, but on the JS side as it would be difficult to implement the same thing in the native code (we couldn't refer to `ReadableStream` and `Headers`).

In that case, I would suggest that `FetchResponse` extends the `NativeResponse` instead of wrapping it. What do you think?

Reply:
> In that case, I would suggest that FetchResponse extends the NativeResponse instead of wrapping it. What do you think?

the `NativeResponse` is just a type `declare class NativeResponse` in typescript. the true implementation is `NetworkFetchModule.NativeResponse()` that provided from jsi.

i got some difficulty to extend `NetworkFetchModule.NativeResponse()`. injecting the js prototype chain could work literally but it's sort of hacky. do you have any better solution to achieve this?

Reply:
I believe this should work:
```typescript
class FetchResponse extends NetworkFetchModule.NativeResponse {
  // ...
}
```

Reply:
it works. thanks for sharing that.

Reply:
Now I wonder how the docs generation will handle this, but let's not think about that right now 😉 I guess for the docs we can just say it's partially implementing the spec, i.e. we don't need to generate docs for now.